dspace-server-webapp/src/test/java/org/dspace/app/rest/PatchMetadataIT.java

```diff
@@ -0,0 +1,803 @@
+
+import org.dspace.app.rest.builder.CollectionBuilder;
+import org.dspace.app.rest.builder.CommunityBuilder;
+import org.dspace.app.rest.builder.ItemBuilder;
+import org.dspace.app.rest.builder.WorkspaceItemBuilder;
+import org.dspace.app.rest.matcher.MetadataMatcher;
+import org.dspace.app.rest.model.MetadataValueRest;
+import org.dspace.app.rest.model.patch.AddOperation;
+import org.dspace.app.rest.model.patch.MoveOperation;
+import org.dspace.app.rest.model.patch.Operation;
+import org.dspace.app.rest.model.patch.RemoveOperation;
+import org.dspace.app.rest.model.patch.ReplaceOperation;
+import org.dspace.app.rest.test.AbstractEntityIntegrationTest;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.RelationshipType;
+import org.dspace.content.WorkspaceItem;
+import org.dspace.content.service.EntityTypeService;
+import org.dspace.content.service.ItemService;
+import org.dspace.content.service.RelationshipTypeService;
+import org.dspace.content.service.WorkspaceItemService;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MvcResult;
+
+/**
+ * Created by kristof on 20/02/2020
+ */
+public class PatchMetadataIT extends AbstractEntityIntegrationTest {
+
+    @Autowired
+    private RelationshipTypeService relationshipTypeService;
+
+    @Autowired
+    private EntityTypeService entityTypeService;
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private WorkspaceItemService workspaceItemService;
+
+    private Collection collection;
+    private WorkspaceItem publicationItem;
+    private Item personItem1;
+    private Item personItem2;
+    private RelationshipType publicationPersonRelationshipType;
+
+    private List<String> authorsOriginalOrder;
+
+    private String addedAuthor;
+    private String replacedAuthor;
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        context.turnOffAuthorisationSystem();
+
+        Community community = CommunityBuilder.createCommunity(context)
+                .withName("Parent community")
+                .build();
+        collection = CollectionBuilder.createCollection(context, community)
+                .withName("Collection")
+                .build();
+
+        context.restoreAuthSystemState();
+    }
+
+    /**
+     * A method to create a workspace publication containing 5 authors: 3 regular authors and 2 related Person items.
+     * The authors are added in a specific order:
+     * - "Whyte, William": Regular author
+     * - "Dahlen, Sarah": Related Person
+     * - "Peterson, Karrie": Regular author
+     * - "Perotti, Enrico": Regular author
+     * - "Linton, Oliver": Related Person
```

Comment:
It looks (to me) like we are always testing with mixed content...some Entities, and some textual authors.  Could we add a few basic tests to verify this patching also works for *only textual authors*?

Reply:
Another set of test for only textual authors has been included